sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -423,17 +526,35 @@ def create_node(self, node_config, tags, count):
         # Adding the jump pod to the new_nodes list as well so it can be
         # checked if it's scheduled and running along with other pod instances.
         ssh_jump_pod_name = conf['metadata']['labels']['skypilot-ssh-jump']
+        new_nodes_with_jump_pod = new_nodes[:]
         jump_pod = kubernetes.core_api().read_namespaced_pod(
             ssh_jump_pod_name, self.namespace)
-        new_nodes.append(jump_pod)
+        new_nodes_with_jump_pod.append(jump_pod)
+        node_names = [node.metadata.name for node in new_nodes_with_jump_pod]
 
         # Wait until the pods are scheduled and surface cause for error
         # if there is one
-        self._wait_for_pods_to_schedule(new_nodes)
+        logger.info(config.log_prefix +
+                    f'Waiting for pods to schedule. Pods: {node_names}')
+        self._wait_for_pods_to_schedule(new_nodes_with_jump_pod)
         # Wait until the pods and their containers are up and running, and
         # fail early if there is an error
-        self._wait_for_pods_to_run(new_nodes)
+        logger.info(config.log_prefix +
+                    f'Waiting for pods to run. Pods: {node_names}')
+        self._wait_for_pods_to_run(new_nodes_with_jump_pod)
+        logger.info(config.log_prefix +
+                    f'Checking if user in image has sufficient privileges.')
+        self._check_user_privilege(new_nodes)
+        logger.info(config.log_prefix +
+                    f'Setting up SSH in pod.')
+        self._setup_ssh_in_pods(new_nodes)
+        logger.info(config.log_prefix +
+                    f'Setting up environment variables in pod.')
         self._set_env_vars_in_pods(new_nodes)
```

Comment:
Do we need to setup envs for jump pod?

Reply:
The env vars are necessary for pods to interact with GPUs, so it's only necessary for the instance pods, not jump pods.